ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +265,57 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+// Obtains the metadata of a secret.
+const describeSecret = async (res, appid, payload, DB) => {
+    try {
+        // get and check parameter 
+        let secretName = getParam_String(payload, 'secretName')
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        // search secret_metadata 
+        const query = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['secretName', 'description', 'createTime', 'plannedDeleteTime', 'rotationInterval', 'lastRotationDate', 'nextRotationDate'],
+            limit: 1,
+        }
+        let secret_metadata_result = await DB.partitionedFind('secret_metadata', query)
+
+        // build result
+        let result = {}
+        if (secret_metadata_result.docs.length > 0) {
+            const doc = secret_metadata_result.docs[0]
+            result['secretName'] = base64_decode(doc['secretName'])
+            result['description'] = base64_decode(doc['description'])
+            result['createTime'] = doc['createTime']
+            if (doc['plannedDeleteTime']) {
+                result['plannedDeleteTime'] = doc['plannedDeleteTime']
+            }
+            if (doc['rotationInterval']) {
+                result['rotationInterval'] = doc['rotationInterval']
+                result['lastRotationDate'] = doc['lastRotationDate']
+                result['nextRotationDate'] = doc['nextRotationDate']
+            }
+        }
+
```

Comment:
else?

Reply:
There is no else. If it cannot be found, an empty JSON will be returned